src/chapters/orm/studio.rst

```diff
@@ -0,0 +1,116 @@
+
+Finally, in the class declarations of both ``Event`` and ``EventCategory``, extend ``AbstractEntity``.
+
+Move Duplicated Code Into the Superclass
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Open up the ``Event`` class file. Move the 4 class members that we identified as being duplicated across both models up into ``AbstractEntity``. Now your ``Event`` class will *inherit* these members rather than define them itself.
+
+When you move the ``equals`` method up into ``AbstractEntity``, you'll see a compiler error on the last line (this is related to the note we made above). Let's look at this method in detail.
+
+.. sourcecode:: java
+
+   @Override
+   public boolean equals(Object o) {
+      if (this == o) return true;
+      if (o == null || getClass() != o.getClass()) return false;
+      Event that = (Event) o;
+      return id == that.id;
+   }
+
+The compiler error occurs on the last line, and the context menu in IntelliJ describes it for us:
+
+::
+
+   'id' has private access in 'org.launchcode.codingevents.models.AbstractEntity'
+
+
+What is this telling us? As written, ``that`` is an ``Event`` object, since it is the result of casting ``o`` to ``Event`` in the line above: ``event that = (Event) o;``. But when we reference ``that.id`` we are attempting to reference the *private* field ``id``, which lives not in ``Event`` but in ``AbstractEntity``. This is not allowed.
+
+This error is easy to fix; simply change the cast on the next-to-last line to convert ``o`` to an instance of ``AbstractEntity``.
+
+.. sourcecode:: java
+
+   @Override
+   public boolean equals(Object o) {
+      if (this == o) return true;
+      if (o == null || getClass() != o.getClass()) return false;
+      AbstractEntity that = (AbstractEntity) o;
+      return id == that.id;
+   }
+
+.. admonition:: Note
+
+   The ``equals`` method works by checking the value of the identifier/primary key field ``id``. If two objects have the same ``id``, then they should be considered equal. Otherwise, they are not equal.
+
+   It may not be obvious that you can't have an ``Event`` object and an ``EventCategory`` object with the same ``id``. However, the way in which the database manages and assigns these values ensures that won't happen.
+
+Now in your other model, delete the 4 class members that are now inherited from ``AbstractEntity``. You should have NO compiler errors in your application at this point.
+
+The ``@MappedSuperClass`` Annotation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+One final step: Add the ``@MappedSuperClass`` to your ``AbstractEntity`` class. 
+
+This annotation will be discussed in more detail later, but for now you should understand that it ensures that even though ``Event`` and ``EventCategory`` don't have ``id`` fields in their class definition (since they are inherited) these columns will still be stored on the ``event`` and ``event_category`` tables of the database.
```

Comment:
This is a really long sentence.  Try breaking it into smaller chunks, since following the transitions "but", "even though", etc. gets awkward.